trying again

darwyncook opened the pull request:
Don't mess this up Zigster!!!!!!

Zyggy commented:
Well, The Email code does not check anything other than whether or not there is at least one "@" symbol. It needs to check for .org, .edu, .com, or .net. The social security class only checks if there is a "-" in it. It must be in the form of "AAA-BB-CCCC" where they are all numbers with a few other restrictions ("AAA" cannot be 900-999). And for the Hash class, two passwords do not return the same hash.

darwyncook commented:
I added a version 2 to the pull request.

Zyggy commented:
The email code does not check for specifically ".com", ".edu", ".biz", ".gov". Also does not check for punctuation restrictions. The SS class needs to check that "AAA" is not 900-999 or 666. It also can not be "078-05-1120". The hash program does not create two equal hashes for the same password. Also, it does not generate a very difficult hash.

darwyncook commented:
How about this?

import hashlib
import re


class InvalidEmail(Exception):
    pass


class InvalidSocial(Exception):
    pass


class InvalidPassword(Exception):
    pass

"""
class Email:

    def __init__(self, email):
        try:
            a, b = email.split("@")
            z = b.split(".")
            if z[-1] in ["com", "net", "edu"]:
                self.email = email
            else:
                raise InvalidEmail
        except ImportError:
            raise InvalidEmail

    def __str__(self):
        return self.email
"""

class Email:

    def __init__(self, email):
        search_Obj = re.search(r'[a-zA-Z]([a-zA-Z0-9]*)@([a-zA-Z0-9]*)\.edu|biz|com|gov', email, flags=0)
        if search_Obj is not None:
            self.email = search_Obj.group()
        else:
            raise InvalidEmail

    def __str__(self):
        return self.email


class SS:

    def __init__(self, social):
        search_Obj = re.search(r'[0-9]{3}-[0-9]{2}-[0-9]{4}', social, flags=0)
        if search_Obj is not None:
            self.social = search_Obj.group()
        else:
            raise InvalidSocial

    def __str__(self):
        return self.social


class Hash:
    def __init__(self, passwd):
        if len(passwd) > 7:
            self.hash = self.generate_hash(passwd)
        else:
            raise InvalidPassword

    def __eq__(self, other):
        if isinstance(other, str):
            return self.hash == self.generate_hash(other)
        elif isinstance(other, Hash):
            return self.hash == other.hash
        else:
            return NotImplemented

    @staticmethod
    def generate_hash(passwd):
        """ This is NOT secure, a random salt should be added"""
        return hashlib.sha512(passwd.encode("utf-8")).hexdigest()

Zyggy commented:
Is there a reason there are two email classes?

darwyncook commented:
Oops, I should have taken that out. The first email is commented out because I decided to change how I coded it. Some of your classmates were reading my coded instead of unit testing, so I rewrote the code so that it was not readable. :)

Zyggy commented:
This Version seems to pass all of my unit tests. Also, quick question. I got a new laptop and downloaded the 2017 version of python. Will that affect anything or should I just download the version that we have been using and use that instead?

darwyncook commented:
import hashlib
import re


class InvalidEmail(Exception):
    pass


class InvalidSocial(Exception):
    pass


class InvalidPassword(Exception):
    pass


class Email:

    def __init__(self, email):
        searchobj = re.search(r'^[a-zA-Z]([a-zA-Z0-9]*)@[a-zA-Z]([a-zA-Z0-9\.]*)\.(edu|biz|com|gov)$', email, flags=0)
        if searchobj is not None:
            self.email = searchobj.group()
        else:
            raise InvalidEmail

    def __str__(self):
        return self.email


class SS:

    def __init__(self, social):
        search_obj = re.search(r'^(?!219-09-9999|078-05-1120)(?!666|000|9\d{2})\d{3}-(?!00)\d{2}-(?!0{4})\d{4}$',
                               social, flags=0)
        if search_obj is not None:
            self.social = search_obj.group()
        else:
            raise InvalidSocial

    def __str__(self):
        return self.social


class Hash:
    def __init__(self, passwd):
        if len(passwd) > 7:
            self.hash = self.generate_hash(passwd)
        else:
            raise InvalidPassword

    def __eq__(self, other):
        if isinstance(other, str):
            return self.hash == self.generate_hash(other)
        elif isinstance(other, Hash):
            return self.hash == other.hash
        else:
            return NotImplemented

    @staticmethod
    def generate_hash(passwd):
        """ This is NOT secure, a random salt should be added"""
        return hashlib.sha512(passwd.encode("utf-8")).hexdigest()
